### PR TITLE
Document graph cache and bindings for JG clusters

### PR DESCRIPTION
**Pull request by dpitera**
[skip ci]

Issues: #938
Signed-off-by: David Pitera <dpitera@us.ibm.com>

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there an issue associated with this PR? Is it referenced in the commit message?
- [x] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you written and/or updated unit tests to verify your changes?
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [x] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [x] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered?
- [x] If this PR is a documentation-only change, have you added a `[skip ci]`
  tag to the first line of your commit message to avoid spending CPU cycles in
  Travis CI when no code, tests, or build configuration are modified?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by dpitera** [comment]
@analytically @amcp Correct PR :)

Also, let's wait for @pluradj to review

File: docs/multinodejanusgraphcluster.adoc [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+----
+graphs {
+  graph1: conf/graph1.properties,
+  graph2: conf/graph2.properties
+}
+----
+
+then you will access your graphs on the GremlinServer using the fact that the String `graph1` will be bound to the graph opened on the server as per its supplied properties file, and the same holds true for `graph2`.
+
+However, if we use the `ConfiguredGraphFactory` to dynamically create graphs, then those graphs are managed by the <<configuredgraphfactory.adoc#JanusGraphmanager,JanusGraphManager>> and the graph configurations are managed by the <<configuredgraphfactory.adoc#iconfigurationmanagementgraph,ConfigurationManagementGraph>>. This is especially useful because it 1. allows you to define graph configurations post-server-start and 2. allows the graph configurations to be managed in a persisted and distributed nature across your JanusGraph cluster.
```
Comment (pluradj) [inline docs/multinodejanusgraphcluster.adoc]: s/#iconfigurationmanagementgraph/#configurationmanagementgraph

File: docs/multinodejanusgraphcluster.adoc [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+For example, if you update or delete the configuration to a graph on one JanusGraph node, then we must evict that graph from the cache on _every JanusGraph node in the cluster_. Otherwise, we may have inconsistent graph representations across your cluster. JanusGraph automatically handles this eviction using a messaging log queue through the backend system that the graph in question is configured to use.
+
+If one of your servers is configured incorrectly, then it may not be able to successfully remove the graph from the cache.
+
+[IMPORTANT]
+====
+Any updates to your <<configuredgraphfactory.adoc#template-configuration,TemplateConfiguration>> will not result in the updating of graphs/graph configurations previously created using said template configuration. If you want to update the individual graph configurations, you must do so using the <<configuredgraphfactory.adoc#updating-configuration,available update APIs>>. These update APIs will _then_ result in the graphe cache eviction across all JanusGraph nodes in your cluster.
```
Comment (pluradj) [inline docs/multinodejanusgraphcluster.adoc]: s/#updating-configuration/#updating-configurations

**Comment by dpitera** [comment]
@pluradj Repushed

File: docs/multinodejanusgraphcluster.adoc [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+[source, properties]
+----
+hosts: [remoteaddress1.com, remoteaddress2.com]
+port: 8182
+username: admin
+password: password
+connectionPool: { enableSsl: true }
+serializer: { className: org.apache.tinkerpop.gremlin.driver.ser.GryoMessageSerializerV3d0,
+              config: { ioRegistries: [org.apache.tinkerpop.gremlin.tinkergraph.structure.TinkerIoRegistryV3d0] }}
```
Comment (pluradj) [inline docs/multinodejanusgraphcluster.adoc]: This should match what's in the JanusGraph version of [`remote-objects.yaml`](https://github.com/JanusGraph/janusgraph/blob/master/janusgraph-dist/src/assembly/static/conf/remote-objects.yaml#L3):

```
serializer: { className: org.apache.tinkerpop.gremlin.driver.ser.GryoMessageSerializerV1d0,
             config: { ioRegistries: [org.janusgraph.graphdb.tinkerpop.JanusGraphIoRegistry] }}
```

**Comment by chupman** [comment]
Not directly part of this PR, but for the example in 8.10 I think it would be a lot better to have the remote connect be sessioned

:remote connect tinkerpop.server conf/remote.yaml session

When I went through the docs and copy and pasted the commands line breaks were added. Using a session would give new users a much better experience. Also I'm going to open an issue to add info on sessions to 7.1.1.1 since they seem to be a gap in the current documentation.

**Comment by chupman** [comment]
Also in the 8.10 gremlin examples there are 4 spots with Map map = new HashMap(), that should instead be map = new HashMap()

**Pull request merged**